src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -878,66 +1054,124 @@ else if (block instanceof BlockEndPortal || block instanceof BlockMobSpawner ||
             return false;
         }
 
-        final int slot = worker.findFirstSlotInInventoryWith(stack.getItem());
+        final int slot = worker.findFirstSlotInInventoryWith(stack.getItem(), stack.getItemDamage());
         if (slot != -1)
         {
             getInventory().decrStackSize(slot, 1);
-            reduceNeededResources(block);
+            reduceNeededResources(stack);
         }
+
+        if(secondaryItem == null)
+        {
+            return true;
+        }
+
+        final int secondarySlot = worker.findFirstSlotInInventoryWith(secondaryItem, 0);
+        if (secondarySlot != -1)
+        {
+            getInventory().decrStackSize(secondarySlot, 1);
+            reduceNeededResources(new ItemStack(secondaryItem, 1));
+        }
+
         return true;
     }
 
     /**
      * Reduces the needed resources by 1.
      *
-     * @param block the block which has been used now.
+     * @param stack the stack which has been used now.
      */
-    public void reduceNeededResources(final Block block)
+    public void reduceNeededResources(final ItemStack stack)
     {
         final AbstractBuilding workerBuilding = this.getOwnBuilding();
         if (workerBuilding instanceof BuildingBuilder)
         {
-            ((BuildingBuilder) workerBuilding).reduceNeededResource(block, 1);
+            ((BuildingBuilder) workerBuilding).reduceNeededResource(stack, 1);
         }
     }
 
-    private void spawnEntity(@Nullable final Entity entity)
+    /**
+     * Get the entity of an entityInfo object.
+     * @param entityInfo the input.
+     * @return the output object or null.
+     */
+    @Nullable
+    private Entity getEntityFromEntityInfoOrNull(Template.EntityInfo entityInfo)
     {
-        if (entity != null && job instanceof JobBuilder)
+        try
+        {
+            return EntityList.createEntityFromNBT(entityInfo.entityData, world);
+        }
+        catch (Exception e)
         {
-            final BlockPos pos = ((JobBuilder) job).getStructure().getOffsetPosition();
+            Log.getLogger().info("Couldn't restore entitiy", e);
+            return null;
+        }
+    }
 
-            if (entity instanceof EntityHanging)
+    private Boolean spawnEntity(@NotNull final Structure.StructureBlock currentBlock)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "spawnEntity" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)